File: csharp/perfect-numbers/PerfectNumbers.cs
using System;
using System.Linq;
using System.Collections;
using System.Collections.Generic;

public enum Classification
{
    Perfect,
    Abundant,
    Deficient
}

public static class PerfectNumbers
{
    public static Classification Classify(int number)
    {
        if (number < 1)
            throw new ArgumentOutOfRangeException();

        int sumOfFactors = GetFactors(number).Sum();

        if (sumOfFactors == number)
            return Classification.Perfect;
        else if (sumOfFactors > number)
            return Classification.Abundant;
        else
            return Classification.Deficient;
    }

    public static IEnumerable<int> GetFactors(int number)
    {
        List<int> factors = new List<int>();

        for (int i = 1; i <= number/2; i++)
            if (number%i==0)
                factors.Add(i);

        return factors;
    }
}

File: python/perfect-numbers/perfect_numbers.py
def classify(number):
    if number < 1:
        raise ValueError('the Number must be greater than 1.')

    summary = sum(get_factors(number))

    if summary == number:
        return 'perfect'
    elif summary > number:
        return 'abundant'
    else:
        return 'deficient'

def get_factors(number):
    return [i for i in range(1, number//2 + 1) if number%i == 0]
